lambdas/select-cri/src/main/java/uk/gov/di/ipv/core/selectcri/SelectCriHandler.java

```diff
@@ -0,0 +1,106 @@
+import uk.gov.di.ipv.core.library.domain.JourneyResponse;
+import uk.gov.di.ipv.core.library.exceptions.HttpResponseExceptionWithErrorBody;
+import uk.gov.di.ipv.core.library.helpers.ApiGatewayResponseGenerator;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+import uk.gov.di.ipv.core.library.helpers.RequestHelper;
+import uk.gov.di.ipv.core.library.service.ConfigurationService;
+import uk.gov.di.ipv.core.library.service.IpvSessionService;
+import uk.gov.di.ipv.core.library.service.UserIdentityService;
+
+import java.util.List;
+
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.FRAUD_CRI_ID;
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.KBV_CRI_ID;
+import static uk.gov.di.ipv.core.library.config.ConfigurationVariable.PASSPORT_CRI_ID;
+
+public class SelectCriHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final String CRI_START_JOURNEY = "/journey/cri/start/%s";
+    public static final String JOURNEY_ERROR = "/journey/error";
+
+    private final ConfigurationService configurationService;
+    private final UserIdentityService userIdentityService;
+    private final IpvSessionService ipvSessionService;
+
+    public SelectCriHandler(
+            ConfigurationService configurationService,
+            UserIdentityService userIdentityService,
+            IpvSessionService ipvSessionService) {
+        this.configurationService = configurationService;
+        this.userIdentityService = userIdentityService;
+        this.ipvSessionService = ipvSessionService;
+    }
+
+    @ExcludeFromGeneratedCoverageReport
+    public SelectCriHandler() {
+        this.configurationService = new ConfigurationService();
+        this.userIdentityService = new UserIdentityService(configurationService);
+        this.ipvSessionService = new IpvSessionService(configurationService);
+    }
+
+    @Override
+    @Tracing
+    @Logging(clearState = true)
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent event, Context context) {
+        LogHelper.attachComponentIdToLogs();
+        try {
+            String ipvSessionId = RequestHelper.getIpvSessionId(event);
+            String userId = ipvSessionService.getUserId(ipvSessionId);
+            List<String> visitedCredentialIssuers =
+                    userIdentityService.getUserIssuedCredentialIssuers(userId);
+
+            String passportCriId = configurationService.getSsmParameter(PASSPORT_CRI_ID);
+            String fraudCriId = configurationService.getSsmParameter(FRAUD_CRI_ID);
+            String kbvCriId = configurationService.getSsmParameter(KBV_CRI_ID);
+            String addressCriId = configurationService.getSsmParameter(KBV_CRI_ID);
+
+            if (userHasNotVisited(visitedCredentialIssuers, passportCriId)) {
```

Comment:
I think the ordering of these if statements are now going to control the order that the CRI's are visited.

We will probably need to order these so that the journey doesn't change unexpectedly?

Reply:
Done